src/app/clusters/software-diagnostics-server/software-diagnostics-server.cpp

```diff
@@ -63,22 +63,27 @@ CHIP_ERROR SoftwareDiagosticsAttrAccess::Read(const ConcreteReadAttributePath &
 
     switch (aPath.mAttributeId)
     {
-    case CurrentHeapFree::Id: {
+    case CurrentHeapFree::Id:
         return ReadIfSupported(&DiagnosticDataProvider::GetCurrentHeapFree, aEncoder);
-    }
-    case CurrentHeapUsed::Id: {
+    case CurrentHeapUsed::Id:
         return ReadIfSupported(&DiagnosticDataProvider::GetCurrentHeapUsed, aEncoder);
-    }
-    case CurrentHeapHighWatermark::Id: {
+    case CurrentHeapHighWatermark::Id:
         return ReadIfSupported(&DiagnosticDataProvider::GetCurrentHeapHighWatermark, aEncoder);
-    }
-    case ThreadMetrics::Id: {
+    case ThreadMetrics::Id:
         return ReadThreadMetrics(aEncoder);
+    case Clusters::Globals::Attributes::FeatureMap::Id: {
+        BitFlags<SoftwareDiagnosticsFeature> features;
+
+        if (DeviceLayer::GetDiagnosticDataProvider().SupportsWatermarks())
+        {
+            features.Set(SoftwareDiagnosticsFeature::kWaterMarks);
+        }
+
+        return aEncoder.Encode(features);
```

Comment:
Please add this to the `attributeAccessInterfaceAttributes` in `src/app/zap-templates/zcl/zcl.json` (right next to "ThreadMetrics" there), so we can hopefully stop allocating RAM for it.

Reply:
added featuremap to that list. Ran zap regen though and saw no differences in output.